### PR TITLE
[OC-1602] Delete cards endpoint to delete multiple cards by date criteria

### PR DESCRIPTION
**Pull request by quinarygio**
[OC-1602] Delete cards endpoint to delete multiple cards by date criteria

[OC-1602]: https://opfab.atlassian.net/browse/OC-1602

### CODE REVIEW

File: src/test/api/karate/cards/deleteCardsByEndDate.feature [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    And header Authorization = 'Bearer ' + authTokenAsTSO
+    When method get
+    Then status 404
+
+#get card with user operator1 
+    Given url opfabUrl + 'cards/cards/api_test.process2card2'
+    And header Authorization = 'Bearer ' + authTokenAsTSO
+    When method get
+    Then status 200
+
```
Comment (freddidierRTE) [inline src/test/api/karate/cards/deleteCardsByEndDate.feature:93]: Delete the remaining cards to clean cards mongo collection for next tests

File: services/core/cards-publication/src/test/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessServiceShould.java [hunk 2]
```diff
@@ -1107,4 +1108,48 @@ void processCardWithEntitiesRequiredToRespondNotSubsetOfEntitiesAllowedToRespond
...
+    void deleteCards_by_endDate() {
+
+        EasyRandom easyRandom = instantiateRandomCardGenerator();
+        int numberOfCards = 10;
+
+        List<CardPublicationData> cards = instantiateSeveralRandomCards(easyRandom, numberOfCards);
+        
+        Instant ref = Instant.now();
+        AtomicInteger i = new AtomicInteger(1);
+        cards.forEach(c -> {
```
Comment (freddidierRTE) [inline services/core/cards-publication/src/test/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessServiceShould.java:1122]: Why not using just int ?

File: services/core/cards-publication/src/main/modeling/swagger.yaml [hunk 3]
```diff
@@ -731,6 +731,23 @@ paths:
           description: created
           schema:
             $ref: '#/definitions/CardCreationReport'
+    delete:
+      tags:
+        - cards
+        - deletion
+      summary: delete cards
+      description: Delete cards before specified date from OperatorFabric
```
Comment (freddidierRTE) [inline services/core/cards-publication/src/main/modeling/swagger.yaml:739]: Add explanation about the behaviour : use endDate field and startDate when ... etc..

File: services/core/cards-publication/src/main/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessingService.java [hunk 4]
```diff
@@ -246,6 +246,11 @@ boolean checkIsAllTimeSpanEndDateAfterStartDate(CardPublicationData c) {
         });
     }
 
+    public void deleteCards(Instant endDateBefore) {
+        List<CardPublicationData> cards = cardRepositoryService.findCardByEndDateBefore(endDateBefore);
+        this.deleteCards(cards);
```
Comment (freddidierRTE) [inline services/core/cards-publication/src/main/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessingService.java:251]: I guess it work well when have few cards but if we want to delete thousand of cards , we will parse and load in memory all the card , we shall find a way to delete in mongo via a query without having to load cards

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: services/core/cards-publication/src/test/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessServiceShould.java (hunk 2 above)
Comment (quinarygio) [reply services/core/cards-publication/src/test/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessServiceShould.java:1122]: Variable must be final / effectively final to use it inside lambda.

File: services/core/cards-publication/src/main/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessingService.java (hunk 4 above)
Comment (quinarygio) [reply services/core/cards-publication/src/main/java/org/lfenergy/operatorfabric/cards/publication/services/CardProcessingService.java]: Done

File: services/core/cards-publication/src/main/modeling/swagger.yaml (hunk 3 above)
Comment (quinarygio) [reply services/core/cards-publication/src/main/modeling/swagger.yaml]: Done

File: src/test/api/karate/cards/deleteCardsByEndDate.feature (hunk 1 above)
Comment (quinarygio) [reply src/test/api/karate/cards/deleteCardsByEndDate.feature:106]: Done

File: src/test/api/karate/cards/deleteCardsByEndDate.feature [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	"processInstanceId" : "process2card2",
+	"state": "messageState",
+	"groupRecipients": ["Dispatcher"],
+	"severity" : "COMPLIANT",
+	"startDate" : 1553186770000,
+	"endDate" : 1553186880999,
+	"summary" : {"key" : "defaultProcess.summary"},
+	"title" : {"key" : "defaultProcess.title"},
+	"data" : {"message":"new message (card2) "}
+}
```
[the remark this answers is not in the document]
Comment (quinarygio) [reply src/test/api/karate/cards/deleteCardsByEndDate.feature]: Done

**Pull request merged**